src/dynamics/joint/joint_set.rs

```diff
@@ -93,6 +99,23 @@ impl JointSet {
         ))
     }
 
+    /// Gets a mutable reference to the joint with the given handle without a known generation.
+    ///
+    /// This is useful when you know you want the joint at position `i` but
+    /// don't know what is its current generation number. Generation numbers are
+    /// used to protect from the ABA problem because the joint position `i`
+    /// are recycled between two insertion and a removal.
+    ///
+    /// Using this is discouraged in favor of `self.get_mut(handle)` which does not
+    /// suffer form the ABA problem.
+    pub fn get_unknown_gen_mut(&mut self, i: usize) -> Option<(&mut Joint, JointHandle)> {
+        let (id, handle) = self.joint_ids.get_unknown_gen_mut(i)?;
```

Comment:
This does not have to be mutable:

```suggestion
        let (id, handle) = self.joint_ids.get_unknown_gen(i)?;
```